src/RobotsChecker.ts

```diff
@@ -45,6 +45,7 @@ export default class RobotsChecker {
 
   public async isAllowed(pageUrlTxt: string, userAgent: string): Promise<boolean> {
     const domainRobots = await this.getOrCreateForDomain(new URL(pageUrlTxt));
-    return domainRobots.isAllowed(pageUrlTxt, userAgent);
+    const ret = domainRobots.isAllowed(pageUrlTxt, userAgent);
+    return (typeof ret !== 'undefined') ? ret : true; //default allowed
```

Comment:
I feel we could take the advantage of Typescript here. By defining the model for RobotsChecker module

Reply:
Typescript return types amended [here](https://github.com/web-extractors/arachnid-seo-js/pull/33/files#diff-31b780e9cda5f0e384f33eb09346dc7828c05ad9438e7d2e1832251072b7c2c2R2-R3). the dependency of robots-parser can return undefined, but our RobotsChecker should return only boolean value

Reply:
I think there's a different way to handle it which is the Type Guards in typescript using `terser operators`